Internal(Components): Rename Component[ED-21906]

MuliElementor opened the pull request:
## PR Checklist
<!-- 
Please check if your PR fulfills the following requirements:
**Filling out the template is required.** Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
 -->
- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md


## PR Type
What kind of change does this PR introduce?
<!-- Please check the one that applies to this PR using "x" with no spaces eg: [x]. -->
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

<!--start_gitstream_placeholder-->
### ✨ PR Description
Purpose: Add component renaming functionality to enable users to update component titles through the UI and persist changes to the database.

Main changes:
- Added REST API endpoint `/update-titles` and repository method `update_title()` for bulk component renaming
- Implemented inline editable component names with validation for length, duplicates, and required fields
- Introduced component model methods `getTitle()`, `getComponentName()`, and `getComponentUid()` for dynamic title resolution
- Added custom instance title tracking in component path to display user-defined names over default titles
- Removed unused `is_current_user_allow_to_edit()` method from REST API controller

_Generated by LinearB AI and added by gitStream._
<sub>AI-generated content may contain inaccuracies. Please verify before using.
💡 **Tip:** You can customize your AI Description using **Guidelines** [Learn how](https://docs.gitstream.cm/automation-actions/#describe-changes)</sub>
<!--end_gitstream_placeholder-->

MuliElementor committed:
update ui part

MuliElementor committed:
Merge remote-tracking branch 'origin/main' into ED-21906-rename-components

MuliElementor committed:
Implement component renaming functionality in ComponentsList and update store to handle renaming actions

MuliElementor committed:
Merge: Complete merge with resolved conflicts

MuliElementor committed:
update

MuliElementor committed:
simplify code

MuliElementor committed:
add test for name validation

MuliElementor committed:
add tests for rename

MuliElementor committed:
Refactor error handling in component renaming synchronization and remove unnecessary console logs

MuliElementor committed:
Refactor component title retrieval to ensure it is called after component name change event

MuliElementor committed:
Merge branch 'main' into ED-21906-rename-components

elementorbot committed:
Merge branch 'main' into ED-21906-rename-components

elementorbot committed:
Merge branch 'main' into ED-21906-rename-components

elementorbot committed:
Merge branch 'main' into ED-21906-rename-components

elementorbot committed:
Merge branch 'main' into ED-21906-rename-components

elementorbot committed:
Merge branch 'main' into ED-21906-rename-components

elementorbot committed:
Merge branch 'main' into ED-21906-rename-components

elementorbot committed:
Merge branch 'main' into ED-21906-rename-components

MuliElementor committed:
playwright

MuliElementor committed:
Merge remote-tracking branch 'origin/main' into ED-21906-rename-components

MuliElementor committed:
fix conflict

elementorbot committed:
Merge branch 'main' into ED-21906-rename-components

elementorbot committed:
Merge branch 'main' into ED-21906-rename-components

elementorbot committed:
Merge branch 'main' into ED-21906-rename-components

MuliElementor committed:
fix conflict

elementorbot committed:
Merge branch 'main' into ED-21906-rename-components

elementorbot committed:
Merge branch 'main' into ED-21906-rename-components

elementorbot committed:
Merge branch 'main' into ED-21906-rename-components

elementorbot committed:
Merge branch 'main' into ED-21906-rename-components

elementorbot committed:
Merge branch 'main' into ED-21906-rename-components

elementorbot committed:
Merge branch 'main' into ED-21906-rename-components

elementorbot committed:
Merge branch 'main' into ED-21906-rename-components

elementorbot committed:
Merge branch 'main' into ED-21906-rename-components

elementorbot committed:
Merge branch 'main' into ED-21906-rename-components

elementorbot committed:
Merge branch 'main' into ED-21906-rename-components

elementorbot committed:
Merge branch 'main' into ED-21906-rename-components

elementorbot committed:
Merge branch 'main' into ED-21906-rename-components

MuliElementor committed:
update

MuliElementor committed:
update

MuliElementor committed:
lint

MuliElementor committed:
Refactor: Reorder imports and clean up formatting in components-item.tsx

MuliElementor committed:
fix

elementorbot committed:
Merge branch 'main' into ED-21906-rename-components

MuliElementor committed:
remove

elementorbot committed:
Merge branch 'main' into ED-21906-rename-components

Omerisra6 commented on assets/dev/js/editor/regions/navigator/element.js:
```diff
@@ -115,6 +115,7 @@ export default class extends Marionette.CompositeView {
 		this.listenTo( this.model, 'change', this.onModelChange )
 			.listenTo( this.model.get( 'settings' ), 'change', this.onModelSettingsChange );
 		this.listenTo( this.model, 'change:editor_settings', this.onModelEditorSettingsChange );
+		this.listenTo( this.model, 'change:componentName', this.onComponentNameChange );
```
same

Omerisra6 commented on packages/packages/core/editor-components/src/components/components-tab/components-item.tsx:
```diff
@@ -132,3 +169,49 @@ const addComponentToPage = ( model: DropElementParams[ 'model' ] ) => {
+	}
+
+	return 'none';
+};
```
You added something here? or just moved the inline styles?

Omerisra6 commented on packages/packages/core/editor-components/src/sync/sync-component-rename-to-navigator.ts:
```diff
@@ -0,0 +1,130 @@
+export function initSyncComponentRenameToNavigator() {
+	try {
+		unsubscribe = subscribeWithSelector(
+			( state ) => state[ SLICE_NAME ]?.data,
```
listen to the component name slice

Omerisra6 commented on packages/packages/core/editor-components/src/sync/sync-component-rename-to-navigator.ts:
```diff
@@ -0,0 +1,130 @@
+			context: { componentUid },
+			cause: errorInstance,
+		} );
+	}
```
I think you can remove the error handling in this file entirely as you catch the rror and do nothing

elementorbot committed:
Merge branch 'main' into ED-21906-rename-components

MuliElementor committed:
revert

MuliElementor committed:
Enhance component functionality: Add methods for fetching and updating component titles based on document settings, and improve component model initialization.

MuliElementor committed:
Merge branch 'main' into ED-21906-rename-components

# Conflicts:
#	modules/components/components-rest-api.php
#	packages/packages/core/editor-components/src/api.ts

MuliElementor committed:
Merge remote-tracking branch 'origin/ED-21906-rename-components' into ED-21906-rename-components

MuliElementor committed:
revert

MuliElementor committed:
Refactor component title update logic in Components Repository and REST API

- Simplified the title update process by removing try-catch blocks and directly checking for document existence.
- Renamed the `update_component_title` method to `update_components_title` for consistency.
- Enhanced the validation logic for component names by utilizing a base schema for validation.
- Removed unused error handling code related to component renaming.

This refactor improves code readability and maintainability while ensuring the functionality remains intact.

MuliElementor committed:
revert

MuliElementor committed:
fix

MuliElementor committed:
fix

MuliElementor committed:
fix

MuliElementor committed:
fix

MuliElementor committed:
update

MuliElementor committed:
Enhance sync logic for component renaming in Navigator

- Added `updateElementSettings` to the sync process for component renaming.
- Updated tests to verify the correct behavior of the new settings update.
- Ensured that the component source name is used instead of the title during updates.
- Improved error handling for scenarios where component names match existing names.

This update improves the synchronization of component renaming with the Navigator, ensuring consistency and reliability in the editor's behavior.

MuliElementor committed:
lint

MuliElementor committed:
Merge branch 'main' into ED-21906-rename-components

MuliElementor committed:
update

MuliElementor committed:
lint

MuliElementor committed:
update tests

MuliElementor committed:
fix tests

MuliElementor committed:
Refactor tests for CreateComponentForm to use expect.objectContaining for better assertion clarity. Removed outdated TODO comment from component name validation utility.

MuliElementor committed:
Merge branch 'main' into ED-21906-rename-components

MuliElementor committed:
remove unused content

MuliElementor committed:
update

MuliElementor committed:
update

MuliElementor committed:
update

MuliElementor committed:
Merge branch 'main' into ED-21906-rename-components

MuliElementor committed:
Merge branch 'main' into ED-21906-rename-components

Omerisra6 committed:
Merge branch 'main' into ED-21906-rename-components

Omerisra6 committed:
wip

Omerisra6 committed:
wip

Omerisra6 committed:
wip

Omerisra6 committed:
wip

Omerisra6 committed:
wip

Omerisra6 committed:
wip

Omerisra6 committed:
wip

Omerisra6 committed:
wip

Omerisra6 committed:
wip

Omerisra6 committed:
Merge remote-tracking branch 'elementor/main' into ED-21906-rename-components

Omerisra6 committed:
Merge branch 'main' into ED-21906-rename-components

Omerisra6 committed:
Merge branch 'main' into ED-21906-rename-components

Omerisra6 committed:
Merge branch 'main' into ED-21906-rename-components

ronros-elementor commented on packages/packages/core/editor-components/src/create-component-type.ts:
```diff
@@ -36,6 +38,38 @@ type ContextMenuGroup = {
+	prototype: {
+		initialize: ( attributes: unknown, options: unknown ) => void;
+	};
+};
```
can this maybe be added to the editor canvas package? as this is where all our backbone types seems to be

Omerisra6 committed:
Merge remote-tracking branch 'elementor/main' into ED-21906-rename-components

Omerisra6 committed:
Merge remote-tracking branch 'elementor/ED-21906-rename-components' into ED-21906-rename-components

Omerisra6 committed:
wip

Omerisra6 committed:
Merge branch 'main' into ED-21906-rename-components

Omerisra6 committed:
wip

Omerisra6 committed:
wip
